### ISSUE TITLE
Versions do not move into/out-of group-folder along with the file they belong to

### ISSUE THREAD

**Issue by mdoggydog** [nextcloud/groupfolders#2678]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Preface
This issue may be related to #2343, but I don't really understand the issue being reported there.

### Steps to reproduce
* _Part A_
1. Create a file in a personal folder.
2. Create several versions of the file; name one or more of them for clarity.
3. _Move_ the file into a group-folder.
4. Examine the versions of the file in the sidebar.
* _Part B_
5. Create versions of the file while it is in the group-folder; name one or more of them for clarity.
6. _Move_ the file back to the personal folder.
7. Examine the versions of the file in the sidebar.
* _Part C_
8. _Move_ the file back to the group-folder.
9. Examine the versions of the file in the sidebar.

### Expected behaviour
* The versions displayed in Step 4 should match those existing in Step 2.
* The versions displayed in Step 7 should be a superset of those created in Step 2 and Step 5.
* The versions displayed in Step 9 should match those visible in Step 7.

In other words, the versions of a file should follow the file around wherever it goes.

### Actual behaviour
* Only the current version is displayed in Step 4.
* The versions originally created in Step 2 reappear in Step 7.
* The versions created in Step 5 have vanished in Step 7.
* The versions created in Step 5 reappear in Step 9; those created in Step 2 disappear again.

In other words:  instead of the versions being attached to and moving around with the file, it appears that the file has one set of versions that live in the personal folder, and a separate set that lives in the groups-folder.  Furthermore, when the file is moved out of a folder, it leaves a set of versions behind (that may never be seen again).

### Server configuration

**Operating system:** Debian Bookworm

**Web server:** Apache 2.4.57

**Database:** Postgresql 15.5

**PHP version:** 8.2.7

**Nextcloud version:** 27.1.4

**Group folders version:** 15.3.1

**Updated from an older Nextcloud/ownCloud or fresh install:** updated

**Where did you install Nextcloud from:** updater.phar

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP

### Client configuration
**Browser:** various

**Operating system:** various

#### Web server error log
_No web server errors_

#### Nextcloud log (data/nextcloud.log)
Not clear there were any log entries related to this.

#### Browser log
_No browser errors_

**Comment by joshtrichards** [comment]
Should have gotten better after #2910. Need to test it, if someone wishes to report back. :)

**Comment by XueSheng-GIT** [comment]
@joshtrichards definitely got better. Still, there's some room for improvement.

Two issues I still see (tested on NC 29.0.7, groupfolders 17.0.4):

1. Moving a file with versions from regular storage to a groupfolder: only preview of latest version is shown in the sidebar. All other previews are broken (but versions are available). Once the file has been moved back to regular storage, the previews are shown again.

2. Moving a file with versions from regular storage to a groupfolder, renaming that file and moving it back again to regular storage: Only initial and current versions are available, all others are gone (also labled versions). But renaming that file to its original name makes the versions re-appear.

I can do a screen recording if required.

**Comment by mdoggydog** [comment]
@joshtrichards, a follow-up:

tl;dr: "unfortunately, still quite broken"

Tested on NC 30.0.4 with Group Folders 18.0.8:
 * Part A, Step 4 (move from regular to group)
   * Good (improvement): The versions created in regular storage are now still available when the file has been moved to a group folder.
   * Bad: As @XueSheng-GIT remarked, previews for all versions, except for the current version, are broken.
 * Part B, Step 7 (move back to regular):
   * Bad: Except for the current version, all versions created in group folder are missing, as are _unnamed_ versions which had been created in regular storage originally in Step 2.
 * Part C, Step 9 (move back to group again):
   * Good (but same as before):  Group-folder-created versions reappear when file is moved back to group folder.  (The original unnamed regular-storage-created versions reappear also.)
   * Bad:  If new versions are made in regular storage at Step 7, they __disappear__ when the file is moved back to group folder.  In fact, even the Current version is not shown in the version listing in the sidebar!  (However, opening the file does still yield the current version.)

Without looking at any code, it feels like there is now a one-time, one-way move (or copy?) of versions from regular-storage to group-folder.

Something similar happens if the converse test is performed, i.e.:
  * Create file in group-folder; create G1 version.
  * Move file to regular-storage; create R2 version.
     * (G1 still available in regular-storage.)
  * Move file back to group-folder; create G3 version.
     * (G1 and R2 still available in group-folder.)
  * Move file back to regular-storage; create R4 version.
     * _G3 version not available in regular-storage._  (G1 and R2 still available.)
  * Move file back to group-folder.
     * _R4 version not available in group-folder._  (But, G3 is back, and G1/R2 still available.) 

...and that is all the permutations that I tried this time around.

I'm kind of curious why this is so much more complicated than "move files from DirectoryA to DirectoryB"....
